zipkin-lens/src/test/data/malformed.json

```diff
@@ -0,0 +1,82 @@
+[
+  {
+    "traceId": "6cf5624e3452d1bc202a57dad35921c0",
+    "parentId": "d65495313f944801",
+    "id": "d5cb70daead478dc",
+    "name": "kong.header_filter",
+    "timestamp": 1570005203864000
+  },
+  {
+    "traceId": "6cf5624e3452d1bc202a57dad35921c0",
+    "parentId": "d65495313f944801",
+    "id": "80b8efd52d66af27",
+    "name": "kong.body_filter",
+    "timestamp": 1570005203864000
+  },
+  {
+    "traceId": "6cf5624e3452d1bc202a57dad35921c0",
+    "parentId": "ed395cb866d02665",
+    "id": "2a49858b1f1d506b",
+    "name": "kong.rewrite",
+    "timestamp": 1570005203829000
+  },
+  {
+    "traceId": "6cf5624e3452d1bc202a57dad35921c0",
+    "parentId": "d65495313f944801",
+    "id": "3e53779bdde7c111",
+    "name": "kong.access",
+    "timestamp": 1570005203829000,
+    "duration": 16000
+  },
+  {
+    "traceId": "6cf5624e3452d1bc202a57dad35921c0",
+    "parentId": "ed395cb866d02665",
+    "id": "d65495313f944801",
+    "kind": "CLIENT",
+    "name": "kong.proxy",
```

Comment:
bad span name. prefer route-based ex get /users/{userId} or even get as kong isn't helpful

what you are describing here, seems to be better as a tag

Reply:
the client span should be semantically correct for http. It is better to have a coherent client span than any of the other spans mentioned. As you noticed, they didn't render and actually made the diagram more confusing vs if they didn't exist at all. My advice is probably use one client span and render the rest with annotations if it is commonly the case you cannot record a duration of precision less than 1ms.